lighthouse-core/report/html/renderer/element-screenshot-renderer.js

```diff
@@ -0,0 +1,97 @@
+/**
+ * @license
+ * Copyright 2019 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+'use strict';
+
+/**
+ * @fileoverview This file dislays a screenshot of the page with a particular element being
+ * highlighted.
+ */
+
+/* globals self  */
+
+class ElementScreenshotRenderer {
+  static renderClipPath(dom, clipId, {top, bottom, left, right}) {
+    const clipPathSvg = dom.createElement('div');
+    clipPathSvg.innerHTML = `<svg height="0" width="0">
+        <defs>
+          <clipPath id='${clipId}' clipPathUnits='objectBoundingBox'>
+            <polygon points="0,0  1,0  1,${top} 0,${top}" ></polygon>
+            <polygon points="0,${bottom} 1,${bottom} 1,1 0,1" ></polygon>
+            <polygon points="0,${top} ${left},${top} ${left},${bottom} 0,${bottom}" ></polygon>
+            <polygon points="${right},${top} 1,${top} 1,${bottom} ${right},${bottom}" ></polygon>
+          </clipPath>
+        </defs>
+      </svg>`;
+    return clipPathSvg;
+  }
+
+  /**
+   * @param {DOM} dom
+   * @param {ParentNode} templateContext
+   * @param {LH.Audit.Details.NodeValue} item
+   * @param {string} fullpageScreenshotUrl
+   * @return {Element}
+   */
+  static render(dom, templateContext, item, fullpageScreenshotUrl) {
+    const tmpl = dom.cloneTemplate('#tmpl-lh-element-screenshot', templateContext);
+    const previewContainer = dom.find('.lh-element-screenshot', tmpl);
+
+    // todo: support desktop screenshots
```

Comment:
ideally just passed around along with the `fullPageScreenshot` artifact so there's no support for specific screen sizes :)

Reply:
Added support for desktop screenshots now, i.e. screenshots that are wider than the element screenshot display area.

Right now if there's a big element highlighted on the screenshot it'll just be cut off with `overflow: hidden`. Maybe in the future we could address that and maybe zoom out a bit? Probably not something that needs to be part of v1.

Reply:
Gotcha yeah let's punt on that for v1 :)